samtranslator/model/sam_resources.py

```diff
@@ -2237,6 +2245,93 @@ def _construct_appsync_api_resources(
+                )
+
+            name, auth_dict = self._validate_auth_type_and_config(auth, index)
+
+            # The first index in authentication is the primary authentication
+            if index == 0:
+                api.AuthenticationType = auth.Type
+                if name:
+                    setattr(api, name, auth_dict)
+                continue
+
+            # All auths after the first are put into AdditionalAuthenticationProviders
+            additional_auth: AdditionalAuthenticationProviderType = {"AuthenticationType": auth.Type}
+            if name and auth_dict:
+                additional_auth[name] = auth_dict
+
+            additional_auths.append(additional_auth)
+
+        if additional_auths:
+            api.AdditionalAuthenticationProviders = additional_auths
+
+    def _validate_auth_type_and_config(
+        self, auth: aws_serverless_graphqlapi.Auth, index: int
+    ) -> Tuple[
+        Optional[Literal["LambdaAuthorizerConfig", "OpenIDConnectConfig", "UserPoolConfig"]],
+        Optional[
+            Union[LambdaAuthorizerConfigType, OpenIDConnectConfigType, UserPoolConfigType, CognitoUserPoolConfigType]
+        ],
+    ]:
+        """
+        Validates the authentication type and returns the name of the config property and the respective dictionary.
+
+        This function will throw an error if auth.Type is not a recognized value.
+        """
+        if auth.Type == "API_KEY" or auth.Type == "AWS_IAM":
+            return None, None
+
+        if auth.Type == "AWS_LAMBDA":
+            lambda_authorizer = sam_expect(
+                auth.LambdaAuthorizer, self.logical_id, f"Auth.{index}.LambdaAuthorizer"
+            ).to_not_be_none("'LambdaAuthorizer' must be defined if type is 'AWS_LAMBDA'.")
+            return "LambdaAuthorizerConfig", cast(
+                LambdaAuthorizerConfigType, remove_none_values(lambda_authorizer.dict())
+            )
+
+        if auth.Type == "OPENID_CONNECT":
+            openid_connect = sam_expect(
+                auth.OpenIDConnect, self.logical_id, f"Auth.{index}.OpenIDConnect"
+            ).to_not_be_none("'OpenIDConnect' must be defined if type is 'OPENID_CONNECT'.")
+            return "OpenIDConnectConfig", cast(OpenIDConnectConfigType, remove_none_values(openid_connect.dict()))
+
+        if auth.Type == "AMAZON_COGNITO_USER_POOLS":
+            user_pool = sam_expect(auth.UserPool, self.logical_id, f"Auth.{index}.UserPool").to_not_be_none(
+                "'UserPool' must be defined if type is 'AMAZON_COGNITO_USER_POOLS'."
+            )
+            if index != 0:
+                # UserPoolConfig does not have the DefaultAction property UNLESS it is the primary authentication
+                # method (first index). If it is an additional authentication, we nullify this value.
+                user_pool.DefaultAction = None
+                return "UserPoolConfig", cast(CognitoUserPoolConfigType, remove_none_values(user_pool.dict()))
```

Comment:
This is confusing and tightly coupled to the expectation it's being called from within a for loop over the Auths property. If my previous refactoring comment is useful, I'd also move this nullification up to the place where you explicitly process the first auth type.

Reply:
If you do not want to refactor as suggested previously, I'd consider renaming the index parameter to something like `is_additional_auth` and using `index != 0` when calling the method.